pynetbox/models/dcim.py

```diff
@@ -13,12 +13,56 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+from six.moves.urllib.parse import urlsplit
+
+from pynetbox.core.query import Request
 from pynetbox.core.response import Record, JsonField
 from pynetbox.core.endpoint import RODetailEndpoint
 from pynetbox.models.ipam import IpAddresses
 from pynetbox.models.circuits import Circuits
 
 
+class TraceableRecord(Record):
+    @property
+    def trace(self):
+        req = Request(
+            key=str(self.id) + "/trace" if not self.url else None,
+            base=self.endpoint.url,
+            token=self.api.token,
+            session_key=self.api.session_key,
+            http_session=self.api.http_session,
+        )
+        ret = []
+        for (termination_a_data, cable_data, termination_b_data) in req.get():
+            this_hop_ret = []
+            for hop_item_data in (termination_a_data, cable_data, termination_b_data):
+                # if not fully terminated then some items will be None
+                if not hop_item_data:
+                    this_hop_ret.append(hop_item_data)
+                    continue
+
+                url_path = urlsplit(hop_item_data["url"]).path
+                if url_path.startswith("/api/dcim/cables"):
+                    return_obj_class = Cables
+                elif url_path.startswith("/api/dcim/front-ports"):
+                    return_obj_class = FrontPorts
+                elif url_path.startswith("/api/dcim/interfaces"):
+                    return_obj_class = Interfaces
+                elif url_path.startswith("/api/dcim/rear-ports"):
+                    return_obj_class = RearPorts
+                else:
+                    raise NotImplementedError(
+                        "unable to unpack item data from endpoint '{}'".format(url_path)
+                    )
+                this_hop_ret.append(
+                    return_obj_class(hop_item_data, self.endpoint.api, self.endpoint)
```

Comment:
You _should_ be able to just supply the `hop_item_data` and it'll figure out the rest of what it needs to based on the value of `url`.

Reply:
Ahh.. nice, I'll try that.

Reply:
I tried, it wants args though :/

```
Traceback (most recent call last):
  File "./test.py", line 33, in <module>
    main()
  File "./test.py", line 26, in main
    trace = iface.trace()
  File "/home/raddessi/dev/spotx/sysops/python-modules/spotx_sysops/src/pynetbox/models/dcim.py", line 57, in trace
    return_obj_class(hop_item_data)
TypeError: __init__() missing 2 required positional arguments: 'api' and 'endpoint'
```